tests/python/pants_test/build_graph/test_build_graph.py

```diff
@@ -247,19 +248,18 @@ def inject_address_closure(self, spec):
     self.build_graph.inject_address_closure(Address.parse(spec))
 
   def test_invalid_address(self):
-    with self.assertRaisesRegexp(AddressLookupError, '^.* does not contain any BUILD files.$'):
+    with self.assertRaisesRegexp(AddressLookupError, '.* does not contain any BUILD files.'):
       self.inject_address_closure('//:a')
 
     self.add_to_build_file('BUILD',
                            'target(name="a", '
                            '  dependencies=["non-existent-path:b"],'
                            ')')
-    with self.assertRaisesRegexp(BuildGraph.TransitiveLookupError,
-                                 '^.*/non-existent-path does not contain any BUILD files.'
-                                 '\s+when translating spec non-existent-path:b'
-                                 '\s+referenced from //:a$'):
+    with self.assertRaisesRegexp(AddressLookupError,
+                                 '.*non-existent-path.*does not contain any BUILD files.'):
       self.inject_address_closure('//:a')
 
+  @unittest.skip(reason='TODO: #4515')
```

Comment:
Do we have a plan to pick this up soon?

Regressing usability feels bad...

Reply:
Yea, I think that `SelectTransitive` will need to go one way or another... at which point we'll get back our dependency callstack.